Updated README's installation instructions

valeriearoth opened the pull request:
Based on issue #39

valeriearoth committed:
Updated README's installation instructions

Based on issue #39

kwhitehall commented on README.md:
````diff
@@ -16,22 +16,13 @@ If any visitors have insights into achieving this, please see the issue titled
-
-```sbt "run"```
+1. Install Scala: 2.10.x (http://www.scala-lang.org/download/) Be sure to add scala/bin folder to your environment.
+2. Install Spark 1.6.0 with Scala 2.10 dependency. NB: To do so you must build the Spark installation. Download the SciSpark source file at http://spark.apache.org/downloads.html and follow the directions for builidng with Scala 2.11 http://spark.apache.org/docs/latest/building-spark.html#building-for-scala-211
````
wee typo on line 20 @valeriearoth "... follow the directions for builidng" 
Also, please recheck this 2nd pt. It indicates to install spark 1.6.0 with scala 2.10, but build spark with 2.11 (??)

kwhitehall commented on README.md:
```diff
@@ -16,22 +16,13 @@ If any visitors have insights into achieving this, please see the issue titled
+4. Download the latest version of SciSpark from https://github.com/SciSpark/SciSpark
+5. Maven 2.0x+ https://maven.apache.org/download.cgi Be sure to add Maven on your environment.
+6. SBT: v0.13.5+ http://www.scala-sbt.org/download.html
+7. Run sbt clean assembly
```
Can sbt clean assembly  use a different font face so that it is known that this is a command?

valeriearoth committed:
Update README.md

kwhitehall commented:
Thanks @valeriearoth. Can you please update with zeppelin installation stuff?

valeriearoth committed:
Update README.md

chrismattmann commented:
LGTM @kwhitehall @valeriearoth +1 to commit.

kwhitehall commented:
Nice one @valeriearoth!
@chrismattmann we have one more step to go, re adding the SciSpark skin for zeppelin. 
@MBoustani can you please add a README to the https://github.com/SciSpark/scispark_zeppelin_web repo? That way, we can incorporate those instructions here so that the notebook has the SciSpark skin. And thus these notes will cover full installation and infrastructure stack set up. 
Thanks guys!

kwhitehall commented:
@MBoustani the war file in the pom.xml file is no longer available. Success in getting the skin was achieved though by replacing the appropriate files in the webapp folder in the zeppelin-bin installation with those in the src/ folder on the https://github.com/SciSpark/scispark_zeppelin_web 
Also, I could not find those war files on maven ... maybe I'm looking the wrong place?
Is it possible to write a bash script to run after downloading the zeppelin-bin jar to make these copies, and run that as the last part of the installation here? @chrismattmann @BrianWilson1 Thoughts?

chrismattmann commented:
@kwhitehall the WAR file should be a Maven overlay using a WAR dependency @MBoustani I believe you tested this, right? @darth-pr can you have a look since you are Maven WAR master?

kwhitehall commented:
Ack'ed @chrismattmann, will wait on @MBoustani and @darth-pr to re. Providing more details below:
According to the pom.xml file at https://github.com/SciSpark/scispark_zeppelin_web the Maven WAR repo being checked is https://repo.maven.apache.org/maven2/org/apache/zeppelin/ It was in there that I was poking around to see if the path needed fixing in the pom.xml file.

MBoustani commented:
@kwhitehall @chrismattmann as far as I remember Zeppelin folks did not put the war file into Central Maven Repo yet and we should build a war file ourself and use that. That was the problem me and Sujen had. @sujen1412 correct me if I am wrong.

valeriearoth committed:
Update README.md

valeriearoth commented:
I have now added instructions to getting the SciSpark skin and removed the previous "possible pitfalls" as they related to incorrect installation requirements (installing ND4J and ND4S) that have been updated.

chrismattmann commented:
I checked into this @MBoustani RE: the war. FYI:

https://issues.apache.org/jira/browse/ZEPPELIN-712

Also FYI I do not see the WAR file there:
http://repo1.maven.org/maven2/org/apache/zeppelin/zeppelin-zengine/

My suggestion is to include in the README a first step of checking out zeppelin web, and then publishing the WAR file locally before doing so. Or this could be easily automated with a Maven Ant task - @riverma has experience with these e.g., in OODT, a pre Maven module that does something and runs in a multi-module project to install something before a later module dependency. I could set this up too. I'll take a look.

kwhitehall commented:
@chrismattmann this WAR stuff is way beyond me right now. Is the current README file ok still? Specifically step 15. is what will have to be modified once the WAR is sorted out. In the meantime, this is what worked for us. If it's ok for now +1 to merging still?

kwhitehall commented on README.md:
````diff
@@ -16,29 +16,57 @@ If any visitors have insights into achieving this, please see the issue titled
+5. Maven 2.0x+ https://maven.apache.org/download.cgi Be sure to add Maven on your environment.
+6. SBT: v0.13.5+ http://www.scala-sbt.org/download.html
+7. Run ```sbt clean assembly```
+8. Find where your SciSpark.jar (or similarly named) file is and get its path. It is likely under target and then under scala-2.10.
````
Hey @valeriearoth, please consider rewording "It is likely under target and then under scala-2.10." to something that indicates the path e.g. /path_to_SciSpark/target/scala-2.10/SciSpark.jar 
To achieve this for e.g. you can make a SciSpark path in your environment to point to the download of SciSpark as a step, then update to $SCISPARK_HOME/target/scala-2.10

chrismattmann commented:
+1 to merge @kwhitehall we can fix the WAR instructions to be the auto skin later when it's sorted.

MBoustani commented:
@chrismattmann @kwhitehall  I will try it and add the README here.

valeriearoth committed:
Update README.md

valeriearoth committed:
Update README.md

kwhitehall commented:
+1 @chrismattmann 
Nice one @valeriearoth! Thank you so much for this contribution.
